libs/utils.py

```
@@ -0,0 +1,320 @@
+            print(error)
+            return dup_prime_keys
+            # raise Exception(error)
+        else:
+            print('valid..')
+            return []
+    except Exception as e:
+        exc_type, value, tb = sys.exc_info()
+        exception_details = (f"Exception Type: {exc_type} Exception value: {value} "
+                             f"at line no: {tb.tb_lineno}")
+        print(exception_details)
+        raise Exception(e)
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)